Separate correct/incorrect predictions

Erin007 opened the pull request:
Results are now separated by incorrect and correct.  The intent is that it will help students identify scenarios in which the model is inaccurate so they can select better training data to differentiate between the cases where the model is confused. 

https://user-images.githubusercontent.com/12300669/117883668-adb46600-b279-11eb-97dc-82bd146e1732.mov

To accomplish this I refactored `ResultsTable` to take in a composite prop for `results`, hoisting the sorting between correct and incorrect into redux and created a new `ResultsToggle` component.

Erin007 commented on package.json:
```diff
@@ -1,6 +1,6 @@
   "name": "@code-dot-org/ml-playground",
-  "version": "0.0.27",
+  "version": "0.0.28",
   "main": "dist/main.js",
```
hmm I've got some hitch hikers to clean up.

breville commented on src/redux.js:
```diff
@@ -1387,6 +1356,32 @@ export function isUserUploadedDataset(state) {
+  results.predictedLabels = predictedLabels;
+  return results;
+}
+
```
Thanks!  It looks like the new `GenerateResults` is currently avoiding a dependency on anything `ResultsTable`-oriented by reusing the standalone row rendering from `DataTable`, so have at it here!

breville commented:
This currently appears to have lost the ability to scroll the results.

breville commented:
I updated this with a small refactor that lets the Flexbox do its vertical layout again, which reenables scrolling.  I removed the accuracy in the popup, per discussion, but with this refactor it can be shown and the scrolling still works.

Erin007 commented:
> I updated this with a small refactor that lets the Flexbox do its vertical layout again, which reenables scrolling. I removed the accuracy in the popup, per discussion, but with this refactor it can be shown and the scrolling still works.

Thank you!!!
